src/config/toml.rs

```diff
@@ -8,23 +8,85 @@ use crate::config::schema::{Config, ConfigAsset};
 use crate::infra::err;
 use crate::model::asset_name::AssetName;
 
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, PartialEq)]
 pub enum TomlError {
     IO(String),
     Parse(toml::de::Error),
-    Decode,
+    Decode(DecodeError),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum DecodeError {
+    MissingKey {
+        key: String,
+    },
+    InvalidType {
+        key: String,
+        expected: TomlType,
+        found: Value,
+    },
+}
+
+#[allow(dead_code)]
+#[derive(Debug, PartialEq, Eq)]
+pub enum TomlType {
+    String,
+    Integer,
+    Float,
+    Boolean,
+    DateTime,
+    Array,
+    Table,
 }
 
 impl Display for TomlError {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         match self {
             TomlError::IO(e) => write!(f, "[IO Error] {}", e),
             TomlError::Parse(e) => write!(f, "[Parsing Error] {}", e),
-            TomlError::Decode => write!(f, "[Decode Error]"),
+            TomlError::Decode(e) => write!(f, "[Decode Error] {}", e),
+        }
+    }
+}
+
+impl Display for DecodeError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            DecodeError::MissingKey { key } => {
+                write!(f, "'{}' is not configured", key)
```

Comment:
Let's change a wording a bit here
```suggestion
                write!(f, "The key '{}' is missing (a typo or not specified?)", key)
```